### DISCUSSION TITLE
Setting z-index for individual layers

### DISCUSSION THREAD

**Discussion by hazelnutcloud** [StarArawn/bevy_ecs_tilemap#194] [Q&A] [answered]
Is there a way to set z-index for individual layers? I have a tiled map with different layers according to where it's z-index should be relative to the player i.e. above / below. Thank you!

**Comment by hazelnutcloud** [comment, marked as answer]
update: I figured out how to do this. However, after a few seconds, each layer with a z-index higher than my player sprite disappears for some reason.

**Reply by X3CP4o2** [reply]
It would be nice to know what you did

**Comment by frederickjjoubert** [comment]
When inserting the `TilemapBundle` be explicit about the transform's `z` position and make sure that when you set it, it is appropriate to the layering you hope to achieve, as well as your player's sprite `z` position.

```rust
.insert_bundle(TilemapBundle {
            grid_size: TilemapGridSize {
                x: TILE_SIZE as f32,
                y: TILE_SIZE as f32,
            },
            size: tilemap_size,
            storage: tile_storage,
            texture: TilemapTexture(image_handle),
            tile_size,
            transform: Transform::from_xyz(0.0, 0.0, z),
            ..Default::default()
        });
```
